### PR TITLE
Miscellaneous state improvements

### PR DESCRIPTION
**Pull request by cea2aj**
A collection of small state design improvements

These changes include:
- Add UUID to meta slice
- Move searchLoading to the top level of state under the status slice
- Rename state.query.query to state.query.input
- Rename query.latest to query.mostRecentSearch
- Rename vertical.key to vertical.verticalKey

J=SLAP-1664
TEST=auto

Update and add relevant test coverage

### CODE REVIEW

File: src/answers-headless.ts [hunk 1]
```diff
@@ -36,8 +36,8 @@ export default class AnswersHeadless {
     private httpManager: HttpManager,
   ) {}
 
-  setQuery(query: string): void {
-    this.stateManager.dispatchEvent('query/set', query);
+  setQueryInput(input: string): void {
```
Comment (cea2aj) [inline src/answers-headless.ts:39]: I updated this to be consistent with our convention of naming the setters 'setQuery + prop', however I'm not the biggest fan of this naming so I'm considering keeping `setQuery`. I'm not sure if that would make it unclear where the corresponding data is in the state. Let me know your thoughts

Comment (tmeyer2115) [reply src/answers-headless.ts:39]: I like the idea of keeping it `setQuery`. The term `setQueryInput` has a front-end connotation to me. It sounds very much like I'm setting the value of the `<input>` tag for a search bar.

File: src/models/slices/status.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export interface StatusState {
```
Comment (tmeyer2115) [inline src/models/slices/status.ts:1]: I think we should try and come up with a more descriptive name. As a consumer of the library, I would wonder "status of what"?

Comment (cea2aj) [reply src/models/slices/status.ts:1]: maybe `SearchStatusState`?

Comment (tmeyer2115) [reply src/models/slices/status.ts:1]: Sure, that sounds good

Comment (cea2aj) [reply src/models/slices/status.ts]: updated!

File: src/answers-headless.ts (hunk 1 above)
Comment (cea2aj) [reply src/answers-headless.ts]: reverted to `setQuery`

**Review by yen-tt — APPROVED** [review]

**Pull request merged**